pkg/controller/atlasfederatedauth/atlasfederated_auth.go

```diff
@@ -110,8 +117,23 @@ func (r *AtlasFederatedAuthReconciler) ensureIDPSettings(ctx context.Context, fe
 	return workflow.OK()
 }
 
-func federatedSettingsAreEqual(operator, atlas *mongodbatlas.FederatedSettingsConnectedOrganization) bool {
+func federatedSettingsAreEqual(operator, atlas *admin.ConnectedOrgConfig) bool {
 	operator.UserConflicts = nil
 	atlas.UserConflicts = nil
 	return cmp.Diff(operator, atlas) == ""
 }
+
+func GetIdentityProviderForFederatedSettings(ctx context.Context, atlasClient *admin.APIClient, fedSettings *admin.OrgFederationSettings) (*admin.FederationIdentityProvider, error) {
```

Comment:
this looks very different from the implementation before. How did we get the right track here to list identity providers vs. getting just one and to use the Okta ID? was ist some documentation hints?

Reply:
The oktaIdpId was previously used as identity provider ID, in the latest version of the API, the ID is the one generated by atlas. However, in the latest API there's no way to get this ID directly, the only way is list Idps and find the on with the same oktaIdpId